client/components/PetAdoptionForm.js

```diff
@@ -0,0 +1,128 @@
+import React, {useState} from "react"
+
+import ErrorList from "./ErrorsList"
+
+const PetAdoptionForm = (props) => {
+  const [errors, setErrors] = useState({})
+  const [adoption, setAdoption] = useState({
+    name: "",
+    phoneNumber: "",
+    email: "",
+    homeStatus: "",
+    adoptablePetId: props.petId,
+  })
+
+  const handleInputChange = (event) => {
+    setAdoption({
+      ...adoption,
+      [event.currentTarget.id]: event.currentTarget.value
+    })
+  }
+
+  const submitHandler = (event) => {
+    event.preventDefault()
+    if (isValid()) {
+      addPetForAdoption()
+    } 
+  }
+
+  const addPetForAdoption = async () => {
+    try {
+      const response = await fetch("/api/v1/adoptions", {
+        method: "POST",
+        headers: new Headers ({
+          "Content-Type": "application/json",
+        }),
+        body: JSON.stringify(adoption)
+      })
+      if(!response.ok) {
+        const errorMessage = `${response.status} ${response.statusText}`
+        throw new Error(errorMessage)
+      }
+      const responseBody = await response.json()
+      console.log("posted succesfully", responseBody)
+      setErrors({})
+      clearForm()

```

Comment:
I would invoke setErrors from within the `clearForm()` function so we don't have to call it here.  If we are ever going to clear the form it makes sense there should be no errors displayed on the page.

We can also remove line 43 at this point